src/com/company/gamelogger/LogOptions.java

```diff
@@ -0,0 +1,42 @@
+package com.company.gamelogger;
+
+import org.apache.logging.log4j.Level;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class LogOptions {
+    /**
+     * This class the simple configurator for all loggers in application
+     */
+
+    // OFF -> FATAL -> ERROR -> WARN -> INFO -> TRACE -> ALL
+    static final Level LEVEL = Level.INFO;
+
+    // If NOT Empty: On match with regexp, message will accept. Mismatching will deny message.
+    static final String REGEX = "";
+
+    private static final String DATE = new SimpleDateFormat("HH-mm-ss_yyyy.MM.dd").format( new Date());
+
+    static final boolean  WRITE_FILE = false;
+    static final String    FILE_NAME = DATE + ".log";
+    static final String    FILE_PATH = "logs";
+
+    // Layout instructions:
+    //    https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html
+    //    http://logging.apache.org/log4j/2.x/manual/layouts.html
+    //
+    // WARNING. Keys: %c, %l, %M do program's speed very slow!
+
+    // Example of good format of output:
+    //    2019-04-16 22:06:54 [22] gamethread.Main > mainfunc() INFO: The game starts.
+    static final String STDOUT_LAYOUT = "%d{yyyy-MM-dd HH:mm:ss} [%-2.2tid] %-30.30c{2} > %20.20M() %5level: %msg%n%throwable";
```

Comment:
I propose to use just %tid. You never know what numbers can Java threads have. It is very OS-dependent.
I saw many times on Linux the output of "jmap" and it showed me bigger numbers.
The longest which I saw was always 5 digits which makes me think that the maximal thread ID is _probably_ 32767, however I am not sure.
That is why I propose to not predict it at let your OS to write the comlpete thread ID without making it cut.

Reply:
Added 10 digits to tid. If we don't do this, the layout breaks down.